aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -772,6 +772,19 @@ void heaviside_kernel(TensorIterator& iter) {
   });
 }
 
+void ldexp_kernel(TensorIterator& iter) {
+  // Dispatch on self.dtype() (aka, the 0th input of the iter).
+  // We know for a fact that the second argument has dtype that holds
+  // int_same_size_t<scalar_t> because we did the casting in ldexp_out already.
+  AT_DISPATCH_FLOATING_TYPES(iter.dtype(0), "ldexp_cpu", [&]() {
+    cpu_kernel(iter, [](scalar_t mantissa, int_same_size_t<scalar_t> exponent) -> scalar_t {
+      return static_cast<scalar_t>(
+        mantissa * std::pow((scalar_t)(2.0), exponent)
```

Comment:
What are the pros/cons of computing ldexp like this vs. calling std::ldexp? 

Follow-up question: if we compute it as a composite op, what about just computing it without a custom kernel? That is, ldexp = a * 2**b?